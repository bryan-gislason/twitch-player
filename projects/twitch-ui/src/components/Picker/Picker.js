import PropTypes from "prop-types";
import React, { memo } from "react";
import { Picker, StyleSheet, Text } from "react-native";
import { useTheme } from "../../hooks";

const CustomPicker = memo(({ values, style = [], ...props }) => {
  const { colors, typography } = useTheme();

  const rootStyle = {
    fontSize: typography.sizeSecondary,
  };
  const pickerStyle = [
    styles.picker,
    {
      backgroundColor: colors.bodyBackground,
      borderColor: colors.border,
      color: colors.bodyText,
    },
  ].concat(style);

  const pickerItems = Object.keys(values).map(lng => (
    <Picker.Item key={lng} label={values[lng]} value={lng} />
  ));

  return (
    <Text style={rootStyle}>
      <Picker {...props} style={pickerStyle}>
        {pickerItems}
      </Picker>
    </Text>
  );
});

CustomPicker.propTypes = {
  values: PropTypes.object.isRequired,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.arrayOf(PropTypes.object),
  ]),
};

const styles = StyleSheet.create({
  picker: {
    width: "100%",
  },
});

export default CustomPicker;
