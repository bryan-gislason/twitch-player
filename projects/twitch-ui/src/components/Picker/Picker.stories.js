import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react";
import React from "react";
import Picker from ".";

const values = {
  foo: "Foo",
  bar: "Bar",
};

storiesOf("Components|Picker", module)
  .add("without selected value", () => (
    <Picker values={values} onValueChange={action("changed")} />
  ))
  .add("with selected value", () => (
    <Picker
      values={values}
      selectedValue="bar"
      onValueChange={action("changed")}
    />
  ));
