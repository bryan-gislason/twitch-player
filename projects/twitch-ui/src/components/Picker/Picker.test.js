import React from "react";
import { render } from "../../testing";
import Picker from "./Picker";

describe("Picker", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <Picker values={{ foo: "Foo", bar: "Bar" }} testID="picker" />
    );
  });

  it("renders a picker with the specified values", () => {
    expect(instance.getByTestId("picker")).toMatchSnapshot();
  });
});
