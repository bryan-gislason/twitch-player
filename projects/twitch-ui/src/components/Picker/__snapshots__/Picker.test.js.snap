// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker renders a picker with the specified values 1`] = `
<select
  class="rn-fontFamily-14xgk7a rn-fontSize-7cikom rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-width-13qz1uu"
  data-focusable="true"
  data-testid="picker"
  style="background-color: rgb(250, 249, 250); color: rgb(25, 23, 28);"
>
  <option
    value="foo"
  >
    Foo
  </option>
  <option
    value="bar"
  >
    Bar
  </option>
</select>
`;
